log-manager/src/main/java/io/airlift/log/BufferedHandler.java

```diff
@@ -25,20 +29,33 @@
 class BufferedHandler
         extends Handler
 {
+    private static final Logger log = Logger.get(BufferedHandler.class);
     private static final int MAX_BATCH_COUNT = 1024;
     private static final byte[] POISON_MESSAGE = new byte[0];
+    private static final int MIN_WAIT_MILLIS = 100;
+    private static final int MAX_WAIT_MILLIS = 360000;
 
     private final MessageOutput messageOutput;
     private final Thread thread;
+    private Thread retryThread;
     private final BlockingQueue<byte[]> queue = new ArrayBlockingQueue<>(MAX_BATCH_COUNT);
+    private final ArrayList<byte[]> retryBatch = new ArrayList<>(MAX_BATCH_COUNT);
```

Comment:
Seeing a non-threadsafe object like this in an otherwise threadsafe class is highly suspicious for developers. Usually it's better to define it in the context of the loop that needs it where it is obviously only used by a single thread. If it's very very hard to pass the object around in the single threaded context, then you probably want to document this as thread confined.